App.tsx

```diff
@@ -14,7 +14,43 @@ import { AppContainer } from "./src/AppContainer"
 
 import { OutsideClickProvider } from "utils/outsideClick"
 
+import AsyncStorage from "@react-native-async-storage/async-storage"
+import { AppStateProvider } from "./src/state"
+import { AppSettings } from "./src/state"
+
 export default function App() {
+    const [settingsReady, setSettingsReady] = useState(false)
+    const [settings, setSettings] = useState(
+        new AppSettings({ theme: "predefined" })
+    )
+
+    // docs: https://docs.expo.dev/versions/latest/sdk/splash-screen/
+    useEffect(() => {
+        async function prepare() {
+            try {
+                await AsyncStorage.getItem("settings")
+                    .then(settingsJSON => {
+                        if (settingsJSON) {
+                            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                            const parsedSettings: AppSettings =
+                                JSON.parse(settingsJSON)
+                            console.log("loaded theme: " + parsedSettings.theme)
+                            setSettings(settings.copyWith(parsedSettings))
+                        }
+                    })
+                    .catch(err => {
+                        console.log(err)
+                    })
+            } catch (e) {
+                console.warn(e)
+            } finally {
+                setSettingsReady(true)
+            }
+        }
+
+        void prepare()
+    }, [])
+
```

Comment:
Non mischiare `.then` con `async/await`, con quello che hai scritto, l'async await è fondamentalmente inutile e tanto varrebbe scrivere
```ts
    useEffect(() => {
        AsyncStorage.getItem("settings")
            .then(settingsJSON => {
                if (settingsJSON) {
                    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
                    const parsedSettings: AppSettings = JSON.parse(settingsJSON)
                    console.log("loaded theme: " + parsedSettings.theme)
                    setSettings(settings.copyWith(parsedSettings))
                }
            })
            .catch(err => {
                console.log(err)
            })
            .finally(() => {
                setSettingsReady(true)
            })
    }, [])
```

penso che async/await sia più leggibile il più delle volte, ma se lo usi devi assolutamente evitare i `.then`, se no è inutile ahah
```ts
    useEffect(() => {
        async function prepare() {
            try {
                const settingsJSON = await AsyncStorage.getItem("settings")
                if (settingsJSON) {
                    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
                    const parsedSettings: AppSettings = JSON.parse(settingsJSON)
                    console.log("loaded theme: " + parsedSettings.theme)
                    setSettings(settings.copyWith(parsedSettings))
                }
            } catch (e) {
                console.warn(e)
            } finally {
                setSettingsReady(true)
            }
        }

        void prepare()
    }, [])
```

In generale, dove è possibile personalmente sceglierei `async/await` over `.then()` ma ogni tool ha il suo utilizzo, queste due scritture sono completamente equivalenti

Reply:
okay